docs: Multi-port support for v1.17 GA (#19401)
hashicorp/consul: website/content/docs/k8s/multiport/reference/proxyconfiguration.mdx

You can use this field to set up limited access that bypasses the service mesh's mTLS or authorization for health checks or metrics.

Refer to [Envoy proxy advanced bootstrap options](/consul/docs/connect/proxies/envoy#advanced-bootstrap-options) for more information and examples.

#### Values

- Default: None
- Data type: String

### `spec.bootstrapConfig.statsBindAddr`

Specifies that the proxy should expose the `/stats` prefix to the _public_ network at the `ip:port` provided. The IP and port combination must be free within the network namespace where the proxy runs. The IP `0.0.0.0` binds to all available interfaces or a pod IP address if supported by your existing network settings.

#### Values

- Default: None
- Data type: String

### `spec.bootstrapConfig.statsConfigJson`

Specifies a complete [stats config](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-config).

> When provided, this field overrides [`spec.bootstrapConfig.statsTags`](#spec-boostrapconfig-statstags) and enables full control over dynamic tag replacements.

Reviewer: ```suggestion
When provided, this field overrides [`spec.bootstrapConfig.statsTags`](#spec-bootstrapconfig-statstags) and enables full control over dynamic tag replacements.
```

Refer to [Envoy proxy advanced bootstrap options](/consul/docs/connect/proxies/envoy#advanced-bootstrap-options) for more information and examples.

#### Values

- Default: None
- Data type: String

### `spec.bootstrapConfig.statsFlushInterval`

 Configures Envoy's [`stats_flush_interval`](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-flush-interval), which measures the length of the interval between stats sink flushes.

#### Values

- Default: None
- Data type: String

### `spec.bootstrapConfig.statsSinkJson`

> Specifies one or more [stats sinks](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-sinks) to append to sinks defined using [`spec.bootstrapConfig.statsdUrl`](#spec-bootstrapconfig-statsdurl) or [`spec.bootstrapConfig.dogstatsdUrl`](#spec-boostrapconfig-dogstatsdurl).

Reviewer: ```suggestion
Specifies one or more [stats sinks](https://www.envoyproxy.io/docs/envoy/v1.17.2/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-field-config-bootstrap-v3-bootstrap-stats-sinks) to append to sinks defined using [`spec.bootstrapConfig.statsdUrl`](#spec-bootstrapconfig-statsdurl) or [`spec.bootstrapConfig.dogstatsdUrl`](#spec-bootstrapconfig-dogstatsdurl).
```

When this field is specified, you cannot also specify [`spec.dynamicConfig.accessLogs.textFormat`](#spec-dynamicconfig-accesslogs-textformat) in the same configuration.

### `spec.dynamicConfig.accessLogs.path`

Specifies a file output path for the access logs.

#### Values

- Default: None
- Data type: String

### `spec.dynamicConfig.accessLogs.textFormat`

Specifies a format string for the access logs. Refer to [default format string in the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#default-format-string) for more information.

When this field is specified, you cannot also specify [`spec.dynamicConfig.accessLogs.jsonFormat`](#spec-dynamicconfig-accesslogs-jsonformat) in the same configuration.

#### Values

- Default: None
- Data type: String

### `spec.dynamicConfig.accessLogs.type`

Specifies the output type for the access logs.

#### Values

- Default: None
- Data type: String containing one of the following values:

  - `LOG_SINK_TYPE_DEFAULT`
  - `LOG_SINK_TYPE_FILE`
  - `LOG_SINK_TYPE_STDERR`
  - `LOG_SINK_TYPE_STDOUT`

> #### Values
> 
> - Default: None
> - Data type: String
> 

Reviewer: ```suggestion
```
I think this is a copy-paste error?

### `spec.dynamicConfig.exposeConfig`

> Specifies configurations to expose the proxy.

Reviewer: ```suggestion
Specifies configurations for exposing the proxy.
```
Correct?

  | Parameter | Description                                                      | Data type | Default |
  | :-------- | :--------------------------------------------------------------- | --------- | ------- |
  | `nanos`   | Specifies the number of nanoseconds before triggering a timeout. | Integer   | None    |
  | `seconds` | Specifies the number of seconds before triggering a timeout.     | Integer   | None    |

### `spec.dynamicConfig.localConnection.requestTimeout`

Specifies the length of time the proxy is allowed to attempt HTTP requests to the local application instance. Applies to HTTP based protocols only.

#### Values

- Default: None
- Data type: Map that can contain the following parameters:

  | Parameter | Description                                                      | Data type | Default |
  | :-------- | :--------------------------------------------------------------- | --------- | ------- |
  | `nanos`   | Specifies the number of nanoseconds before triggering a timeout. | Integer   | None    |
  | `seconds` | Specifies the number of seconds before triggering a timeout.     | Integer   | None    |

### `spec.dynamicConfig.meshGatewayMode`

> Specifies the mode of operation for how upstreams with a remote destination datacenter are resolved. Refer to [Mesh gateway configuration reference](/consul/docs/connect/proxies/proxy-config-reference#mesh-gateway-configuration-reference) for more information.

Reviewer: ```suggestion
Specifies the proxy's mode of operation for resolving upstreams in remote datacenter destinations. Refer to [Mesh gateway configuration reference](/consul/docs/connect/proxies/proxy-config-reference#mesh-gateway-configuration-reference) for more information.
```
I think this says the same thing in active voice.

#### Values

- Default: None
- Data type: String containing one of the following values:

  - `MESH_GATEWAY_MODE_UNSPECIFIED`
  - `MESH_GATEWAY_MODE_NONE`
  - `MESH_GATEWAY_MODE_LOCAL`
  - `MESH_GATEWAY_MODE_REMOTE`

### `spec.dynamicConfig.mode`

> Specifies the mode the proxy operates in: default, transparent, or direct. Refer to [proxy modes](/consul/docs/connect/proxies/proxy-config-reference#proxy-modes) for more information.

Reviewer: ```suggestion
Configures the proxy to operate in transparent, direct, or default mode. Refer to [proxy modes](/consul/docs/connect/proxies/proxy-config-reference#proxy-modes) for more information.
```

#### Values

- Default: None
- Data type: String containing one of the following values:

  - `PROXY_MODE_DEFAULT`
  - `PROXY_MODE_TRANSPARENT`
  - `PROXY_MODE_DIRECT`

### `spec.dynamicConfig.publicListenerJson`

> Specifies a complete Envoy listener to be delivered in place of the main public listener that the proxy used to accept inbound connections. Refer to [escape-hatch overrides](/consul/docs/connect/proxies/envoy#escape-hatch-overrides) for more information

Reviewer: ```suggestion
Specifies a complete Envoy listener for Consul to deliver in place of the main public listener that the proxy uses to accept inbound connections. Refer to [escape-hatch overrides](/consul/docs/connect/proxies/envoy#escape-hatch-overrides) for more information
```
I could be wrong. I think technically you configure Consul to configure Envoy to do all things, so maybe it should be " . . . for Envoy to deliver . . ."

#### Values

- Default: None
- Data type: String

### `spec.dynamicConfig.transparentProxy`

> Specifies additional configurations for the transparent proxy. Refer to [transparent proxy configuration reference](/consul/docs/connect/proxies/proxy-config-reference#transparent-proxy-configuration-reference) for more information.

Reviewer: ```suggestion
Specifies additional configurations for operating proxies in transparent proxy mode. Refer to [transparent proxy configuration reference](/consul/docs/connect/proxies/proxy-config-reference#transparent-proxy-configuration-reference) for more information.
```

#### Values

- Default: None
- Data type: Map

### `spec.dynamicConfig.transparentProxy.dialedDirectly`

> Determines whether this proxy instance's IP address can be dialed directly by transparent proxies. Transparent proxies typically dial upstreams using the "virtual" tagged address, which load balances across instances. A database cluster with a leader is an example where dialing individual instances can be helpful.

Reviewer: ```suggestion
Determines whether services can dial this proxy instance's IP address directly through transparent proxies. Transparent proxies typically dial upstreams using the _virtual_ tagged address, which load balances across instances. A database cluster with a leader is an example where dialing individual instances can be helpful.
```

#### Values

- Default: `false`
- Data type: Boolean

### `spec.dynamicConfig.transparentProxy.outboundListenerPort`

Specifies the port the proxy listens on for outbound traffic to capture and redirect.

#### Values

- Default: `15001`
- Data type: Integer

## Examples

The following examples demonstrate common ProxyConfiguration CRD configuration patterns for specific use cases.

### Specific use case 

Introduce examples clearly and help readers understand the connections 
between the example snippet and the configuration element you are trying to 
explain. Use phrases such as "In the following example, the `elem` performs 

> some action on a service named `service-name`". 

Reviewer: ```suggestion
### Specific use case 

Introduce examples clearly and help readers understand the connections 
between the example snippet and the configuration element you are trying to 
explain. Use phrases such as "In the following example, the `elem` performs 
some action on a service named `service-name`". 
```
Looks like you still have to write an heading and introduction for this example.
